src/items/itemsTable.component.tsx

```diff
@@ -163,16 +168,28 @@ export function ItemsTable(props: ItemTableProps) {
       {
         header: 'Asset Number',
         Header: TableHeaderOverflowTip,
-        accessorFn: (row) => row.item.asset_number,
+        accessorFn: (row) => row.item.asset_number ?? '',
         id: 'item.asset_number',
+        filterVariant: COLUMN_FILTER_VARIANTS['string'],
+        filterFn: COLUMN_FILTER_FUNCTIONS['string'],
+        columnFilterModeOptions: [
+          ...COLUMN_FILTER_MODE_OPTIONS['string'],
+          ...OPTIONAL_FILTER_MODE_OPTIONS,
+        ],
         size: 250,
         GroupedCell: TableGroupedCell,
       },
       {
         header: 'Purchase Order Number',
         Header: TableHeaderOverflowTip,
-        accessorFn: (row) => row.item.purchase_order_number,
+        accessorFn: (row) => row.item.purchase_order_number ?? '',
         id: 'item.purchase_order_number',
+        filterVariant: COLUMN_FILTER_VARIANTS['string'],
```

Comment:
I like the reuse here. I would personally prefer using `.string` over the array notation as while it can be accessed like an array it isn't really and I think its slightly neater. Also in two minds on the capitalisation, I think I have typically seen such things like `ColumnFilterVariants.STRING` over ` COLUMN_FILTER_VARIANTS.string` e.g. https://stackoverflow.com/questions/838929/naming-why-should-named-constants-be-all-uppercase-in-c-java.

Reply:
Will implement the dot notation. With the capitalisation, is the rationale for the former that the actual constant value in use is the value, not the whole data structure? Therefore, capitalise the key, not the array/dictionary/etc?

Reply:
Yes I believe so. The ColumnFilterVariants is an object containing the constants rather than a constant itself.

Reply:
But then the google style guide currently has https://google.github.io/styleguide/tsguide.html#identifiers-constants. So it looks like the original you had before is preferred after all 😅

Reply:
Ah, it's no problem😅. I didn't even know the google style guide was a resource for styling conventions, so now I know more!

Reply:
Yeah we don't necessarily conform to a specific style guide nor have one ourselves, we decide what we consider to be best practice based on what others do. This is one of those cases, since google uses that it seems sensible to follow it.